src/sync/mod.rs

```diff
@@ -288,13 +288,14 @@ where
             count - 1
         };
         // Check that the tail of anchoring chain is committed to the Bitcoin.
-        let transaction = self.get_transaction(last_index)?;
-        if self.transaction_is_committed(transaction.id())? {
+        let mut transaction = self.get_transaction(last_index)?;
+        let mut status = self.transaction_status(transaction.id())?;
+        if !status.is_unknown() {
```

Comment:
Seems like `is_unknown` is only used with `!`, so maybe it'll be better to have a method like `is_known` or `is_received`?